api/src/main/java/io/minio/MinioClient.java

```diff
@@ -5763,6 +5770,56 @@ public void listenBucketNotification(String bucketName, String prefix, String su
   }
 
 
+
+
+  /**
+   * Listen to bucket notifications. As bucket notification are lazily executed, its required
+   * to iterate.
+   *
+   * </p><b>Example:</b><br>
+   * <pre>{@code
+   * String[] events = {"s3:ObjectCreated:*", "s3:ObjectAccessed:*"};
+   *   try (CloseableIterator<Result<NotificationInfo>> ci = client
+   *     .listenBucketNotification("bucket-name", "", "", events).getMyCloseableIterator()) {
+   *      while (ci.hasNext()) {
+   *        NotificationInfo info = ci.next().get();
+   *        if ( info != null) {
+   *
+   *      System.out.println(info.records[0].s3.bucket.name + "/"
+   *          + info.records[0].s3.object.key + " has been created");
+   *       break;
+   *      }
+   *    }
+   *  }
+   *  }</pre>
+   *
+   * @param bucketName Bucket name.
+   * @param prefix Prefix of concerned objects events.
+   * @param suffix Suffix of concerned objects events.
+   * @param events List of events to watch.
+   *
+   * @return NotificationListener
+   */
+  public NotificationListener listenBucketNotification(String bucketName, String prefix,
+      String suffix, String[] events)
+    throws IOException, InvalidKeyException, NoSuchAlgorithmException, InsufficientDataException,
+    InvalidResponseException, InternalException, NoResponseException, InvalidBucketNameException,
+    XmlPullParserException, ErrorResponseException {
+
+    Multimap<String, String> queryParamMap = HashMultimap.create();
+    queryParamMap.put("prefix", prefix);
+    queryParamMap.put("suffix", suffix);
+    for (String event : events) {
+      queryParamMap.put("events", event);
+    }
+    HttpResponse response = executeReq(Method.GET, getRegion(bucketName),
```

Comment:
you could use `executeGet()`

Reply:
we cant use `executeGet` as it uses a HashMap for `queryParameter`. While using a hashmap the values are overridden in the below code